src/librbd/io/ImageRequest.h

```diff
@@ -99,6 +100,14 @@ class ImageRequest {
     return m_trace;
   }
 
+  bool needs_throttle() {
```

Comment:
These changes don't appear to be required given the comments below

Reply:
I am not sure I understand this correctly. This function is used to check whether this ImageRequest should be throttled. I mean when we got enough token and requeue a request in callback, this request will be NOT needs_throttle(). Then we can dequeue this ImageRequest directly.

Reply:
Yes -- but ```ImageRequestWQ``` shouldn't be signaling in a busy-loop so these aren't needed as part of this PR

Reply:
This is not a busy-loop, the next loop will look at the second item, because the current item is already dequeued by ```ThreadPool::PointerWQ<ImageRequest<I> >::_void_dequeue()```.

So I think this code should be working for global IOPS throttle or seperate throttles.

Reply:
... but it's not needed for PR since there isn't read vs write throttles

Reply:
Hm... yea, that't a preparation for future work. Okey, then I will remove it now and add them when it needed. thanx

Reply:
Sorry Jason, I think I was wrong in my last comment. 

The throttle_flag is necessary in this PR. Because by default, the imageRequest->needs_throttle() should be true and then we need to call iops_throttle->get() to get tokens. But when a request in blockers_list got enough tokens and call the callback to requeue the request into ImageReqeustWQ, we need to make this request bypass the throttle step, so we need to update the throttle_flag to false. as 
https://github.com/ceph/ceph/pull/17032/files#diff-eb33bf11c557d935e95a09c79cd5d1c9R550

If we don't have this flag,  when we requeue the item, the request will call iops_throttle->get() again to get tokens, but actually, we have give enough tokens for this request in add_tokens(). https://github.com/ceph/ceph/pull/17032/files#diff-4aa9c031e18b19bffd1702e3952292ccR476

Reply:
OK -- suggest you rename to ```was_throttled```